src/node_api.cc

```diff
@@ -508,9 +514,22 @@ void napi_module_register_cb(v8::Local<v8::Object> exports,
                              v8::Local<v8::Value> module,
                              v8::Local<v8::Context> context,
                              void* priv) {
+  v8::Isolate* isolate = context->GetIsolate();
   napi_module* mod = static_cast<napi_module*>(priv);
+
+  // Can we be absolutely sure that, if isolate->GetData() returns non-NULL, it
+  // means that the pointer returned is one of ours (i.e. a napi_env)? What if
+  // some other framework makes use of slot 0? There really seems to be a need
+  // for a more robust way of assigning arbitrary data to the isolate.
+  if (isolate->GetData(0) == nullptr) {
```

Comment:
I think we need to solve this without additions to node itself, because we need this to work on version of node that have not been modified so we can ship N-API as a stand-alone package.